docs/upgrading_to_v3.0.md

```diff
@@ -0,0 +1,134 @@
+# Upgrade Guidance
+Before moving forward with adopting components of v3 review the list of breaking changes below. You can find a complete list of features, bug fixes and other updates in the [Changelog](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/CHANGELOG.md).
+
+**Important:** There is no in-place upgrade path from v2 to v3.
+
+## Breaking Changes
+
+- Upgrade minimum required Terraform version to 1.3.0, previously was 0.13.7.
+- Added usage of Remote State data. This feature centralizes terraform variables that are input through all steps in a remote state in 0-bootstrap step. This uses a Google Storage bucket so that you will not have the need to re-declare them while deploying the foundation example. Note that each step requires `read/write` access to this bucket in order to be properly deployed.
+- Added Granular Service Account (SA). In previous versions, only one SAs were used to deploy all steps, this ended up having an excess of permissions given to them. Now, each step has its own SA with very limited permissions.
+- User-specified service accounts in Cloud Build (Bring your Service Account - BYOSA). Using user-specified service accounts in Cloud Build instead of the default Cloud Build service account enables you to grant different permissions to these service accounts depending on the tasks they perform. For more details, see following [link](https://cloud.google.com/build/docs/securing-builds/configure-user-specified-service-accounts).
+- 3-networks directory split into two different directories. The 3-Networks step supported both network modes, Dual Shared VPC and Hub and Spoke. In this release, these two modes have been separated into two different implementations.
+
+**Note:** You can use newer Terraform versions directly on top of already deployed configurations that use older versions. However, note that doing so will prevent you from using these older versions. Also, this will update terraform state and dependencies (external modules and provider's resources). Thus, we recommend to backup your current terraform state before trying to upgrade your codebase.
+
+## Integrating New Features
+
+As stated earlier, there is no direct path for upgrading example foundation codebase. Thus, we discourage trying this by solely applying foundation v3's code (i.e. running `terraform apply` on new code).
+
+In case you require to integrate some of the v3's features, we recommend to review the documentation regarding the feature you are interested in and use v3's code as a guidance for its implementation. We also recommend to deeply review the output from `terraform plan` before applying the updates.
```

Comment:
```suggestion
In case you require to integrate some of the v3's features, we recommend to review the documentation regarding the feature you are interested in and use v3's code as a guidance for its implementation. We also recommend to review the output from `terraform plan` for any destructive operations before applying the updates.
```

Reply:
done